source/common/network/connection_impl.cc

```diff
@@ -588,6 +592,20 @@ void ConnectionImpl::onReadReady() {
     result.action_ = PostIoAction::Close;
   }
 
+  if (result.io_error_.has_value()) {
+    ASSERT(result.action_ == PostIoAction::Close);
+    if (dynamic_cast<ServerConnectionImpl*>(this)) {
+      stream_info_.setConnectionTerminationDetails(kDownstreamConnectionTerminationDetails);
+      stream_info_.setResponseFlag(StreamInfo::ResponseFlag::DownstreamConnectionTermination);
+    } else {
+      stream_info_.setUpstreamTransportFailureReason(kUpstreamConnectionTerminationDetails);
+      stream_info_.setConnectionTerminationDetails(kUpstreamConnectionTerminationDetails);
+      stream_info_.setResponseFlag(StreamInfo::ResponseFlag::UpstreamConnectionTermination);
+    }
```

Comment:
Those values are set on the upstream connection's stream info, and they are never propagated to downstream connection's stream info or access logs. I'm not sure what's the best way to solve that.

Reply:
Could the high level request object query the connection as part of the termination process to determine if termination was graceful or abrupt?